test/backup/backup_test.go

```diff
@@ -0,0 +1,501 @@
+					Expect(err).To(HaveOccurred())
+				})
+
+				It("Should fail when improper format name is used in config", func() {
+					config := getBackupConfigInMap("wrong-prefix")
+
+					configBytes, mErr := json.Marshal(config)
+					Expect(mErr).ToNot(HaveOccurred())
+
+					backup = newBackupWithConfig(backupNsNm, configBytes)
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("name should start with %s", namePrefix(backupNsNm)))
+				})
+
+				It("Should fail when more than 1 cluster is given in backup config", func() {
+					config := getBackupConfigInMap(namePrefix(backupNsNm))
+					aeroCluster := config[common.AerospikeClusterKey].(map[string]interface{})
+					aeroCluster["cluster-two"] = aeroCluster["test-cluster"]
+					config[common.AerospikeClusterKey] = aeroCluster
+
+					configBytes, mErr := json.Marshal(config)
+					Expect(mErr).ToNot(HaveOccurred())
+
+					backup = newBackupWithConfig(backupNsNm, configBytes)
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).To(HaveOccurred())
+				})
+
+				It("Should fail when on-demand backup is given at the time of creation", func() {
+					backup, err = NewBackup(backupNsNm)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup.Spec.OnDemandBackups = []asdbv1beta1.OnDemandBackupSpec{
+						{
+							ID:          "on-demand",
+							RoutineName: "test-routine",
+						},
+					}
+
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).To(HaveOccurred())
+				})
+
+				It("Should fail when non-existing routine is given in on-demand backup", func() {
+					backup, err = NewBackup(backupNsNm)
+					Expect(err).ToNot(HaveOccurred())
+
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup, err = getBackupObj(k8sClient, backup.Name, backup.Namespace)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup.Spec.OnDemandBackups = []asdbv1beta1.OnDemandBackupSpec{
+						{
+							ID:          "on-demand",
+							RoutineName: "non-existing-routine",
+						},
+					}
+
+					err = updateBackup(k8sClient, backup)
+					Expect(err).To(HaveOccurred())
+				})
+
+				It("Should fail when backup service is not present", func() {
+					backup, err = NewBackup(backupNsNm)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup.Spec.BackupService.Name = "wrong-backup-service"
+
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).To(HaveOccurred())
+				})
+
+				It("Should fail when backup service reference is updated", func() {
+					backup, err = NewBackup(backupNsNm)
+					Expect(err).ToNot(HaveOccurred())
+
+					err = CreateBackup(k8sClient, backup)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup, err = getBackupObj(k8sClient, backup.Name, backup.Namespace)
+					Expect(err).ToNot(HaveOccurred())
+
+					backup.Spec.BackupService.Name = "updated-backup-service"
+
+					err = updateBackup(k8sClient, backup)
```

Comment:
can we compare error strings as err returned here, can be due to backup service not available or can not update backup with service name. Check other places as well.